File: Top_Interview_150/Two_Pointers/python/392.py
class Solution:
    def isSubsequence(self, s: str, t: str) -> bool:

        if not s:
            return True

        n = len(s)
        ptr_s = 0
        for c in t:
            if c == s[ptr_s]:
                ptr_s += 1

            if ptr_s == n:
                return True

        return False

File: Top_Interview_150/Two_Pointers/cpp/392.cpp
#include <string>

using namespace std;

class Solution
{
public:
    bool isSubsequence(string s, string t)
    {
        if (s.empty())
            return true;

        int n = s.size();
        int ptr_s = 0;
        for (char c : t)
        {
            if (c == s[ptr_s])
                ptr_s++;

            if (ptr_s == n)
                return true;
        }

        return false;
    }
};